pyproject.toml

```diff
@@ -50,5 +55,24 @@ local_scheme = "no-local-version"
 
 [tool.black]
 
+[tool.ruff]
+select = [
+  "E", # pycodestyle errors
+  "F", # pyflakes errors
+  "I", # isort
+  "ISC", # flake8-implicit-str-concat
```

Comment:
Just to make sure I understand correctly: this will emit an error on something like this:

```py
"foo" "bar"
```

But, by default, it allows this?

```py
x = (
    "foo"
    "bar"
)
```

Reply:
Yes, that's right.

This rule is helps clean up after Black formatting. Black will take stuff like this:

```python
x = (
    "foo"
    "bar"
)
```

And give:

```python
x = "foo" "bar"
```

So ISC will help fix those into a nice:

```python
x = "foobar"
```

Reply:
Sounds great then! I like multiline implicit string concatenation, but agree that implicit string concatenation on a single line is definitely an anti-pattern 👍

Reply:
Black has been planning on fixing this to produce `"foobar"` instead of `"foo" "bar"` in the first place.

It wasn't quite ready for the 2023.0 release, hopefully it will be ready for 2024.0 and we can retire ISC.

Reply:
Does this mean that `ruff` needs to be run after `black`?
In the `.pre-commit-config.yaml` `ruff` is listed before `black`, but I'm not sure what determines the order in which they are executed.

Reply:
For most cases, Ruff comes before Black:

> When running with --fix, Ruff's lint hook should be placed before Ruff's formatter hook, and before Black, isort, and other formatting tools, as Ruff's fix behavior can output code changes that require reformatting.

https://github.com/astral-sh/ruff-pre-commit

Reply:
So `ruff` might make changes that `black` will have to reformat, but `black` will also make changes that `ruff` will need to reformat (the implicit string concatenation mentioned above)?

Reply:
I've never seen it happen, but theoretically, and only once, not an infinite back and forth.